community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/RelationshipDirection.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.newapi;
+
+/**
+ * Low level representation of relationship direction, used to keep traversal state and query tx-state
+ *
+ * Given the graph
+ *      a  -[r1]-> b
+ *      a <-[r2]-  b
+ *      a  -[r3]-> a
+ *
+ * Then
+ *  a.getRelationships( OUTGOING ) => r1
+ *  a.getRelationships( INCOMING ) => r2
+ *  a.getRelationships( LOOP ) => r3
+ *  a.getRelationships( ALL ) => r1, r2, r3
+ *
+ * Note that this contrasts with /** @see #org.neo4j.graphdb.Direction(long), where
+ *  a.getRelationships( org.neo4j.graphdb.Direction.OUTGOING ) => r1, r3
+ *  a.getRelationships( org.neo4j.graphdb.Direction.INCOMING ) => r2, r3
+ *  a.getRelationships( org.neo4j.graphdb.Direction.BOTH ) => r1, r2, r3
+ */
+public enum RelationshipDirection
```

Comment:
Yey, our third direction enum! I'm guessing the `storageengine` direction will go away soon though, or hopefully at least.

Reply:
I don't really see the point of the `storageengine` one, as it's a copy of the `graphdb` direction. This one at least has different (and described) semantics. But yeah, more enums can't be bad, right?